### ISSUE TITLE
Checkbox label not read when navigating to input using arrow keys

### ISSUE THREAD

**Issue by mteffaha** [nvaccess/nvda#5483]
Hello.
Using the setting replicated in the following jsfiddle : 
[jsfiddle link](https://jsfiddle.net/mteffaha/fb7e5vzd/7/)

nvda does not read neither the label nor the aria-labelledby div for the checkbox.
this issue is only existing whith the specific css available in the jsfiddle : 

```css
input[type="checkbox"]{
    float:left;    
}

label{
    display: block;    
}

```
thanks in advance.
Best regards

**Comment by jcsteh** [comment]
In browse mode in Firefox, the label appears just below the checkbox. This is NVDA's normal behaviour with labels; it renders the label where it occurs in the document. Are you not seeing the label at all when you press down arrow after the checkbox? What browser are you using?

**Comment by mteffaha** [comment]
I'm sorry i didn't fully explain the issue at hand . when a bling user is using the checkbox and navigate to it using the tab key the aria-labelledby is read which is the expected behavior . but when the user navigate to it using the arrow keys nvda only reads "checkbox not checked", even though there is an aria-labelledby and a label (it should read at least one of them).
this issue  issue happened when using NVDA 2015.3 along with firefox 42.0 (the issue is not present with neither chrome nor ie).

**Comment by jcsteh** [comment]
Right, but as I said, the label is just after the checkbox, reflecting the author's intent in placing it afterward in the DOM. You'll see the same behaviour with an input type="checkbox" and a label tag.

**Comment by shichongrui** [comment]
The provided jsfiddle has an issue with the wrong id being provided to aria-labelledby. Regardless if you remove that and navigate to the checkbox I can confirm that I'm experiencing the same issue where the checkbox is just read as "checkbox not checked". Then navigating down goes to the label and reads it. If you tab to the checkbox though it will read the label. Is this expected? That tabbing to the checkbox will read the label but navigating to it via the arrow keys will not read the label? In my own testing I seem to have had mix results with this where the arrow keys sometimes reads the label and other times does not.

**Comment by jcsteh** [comment]
As noted [above](https://github.com/nvaccess/nvda/issues/5483#issuecomment-156257956), in browse mode, the label is rendered where it appears in the document. So, if the label is below the control, it will be rendered below the control in browse mode too. A user could press NVDA+tab to query the focus, though, in which case the label would be read.
